Fix occasional panic when dart stacktraces are not present sometimes

## Summary

Certain dart crashes do not have exceptions and/or frames in it. Handle these safely without panic.

## Details

```
panic: runtime error: index out of range [0] with length 0

goroutine 164 [running]:
backend/api/event.EventField.NeedsSymbolication({{0x5c, 0xfa, 0xd0, 0x48, 0x6e, 0x69, 0x45, 0x64, 0x88, 0xbf, ...}, ...})
	/go/api/event/event.go:1283 +0x785
backend/ingest-worker/measure.processIngestBatchSync({0x5043420, 0x3b9d2fad0820}, {{0x3b9d3278cde0, 0x24}, {0x3b9d3278ce10, 0x24}, {0x3b9d3278ce40, 0x24}, {0x3b9d32248160, 0x3}, ...})
	/go/src/measure/event.go:1137 +0x765
backend/ingest-worker/measure.ConsumeHandler({0x5043420, 0x3b9d2fad0820}, {0x3b9d32c80000, 0x1a067, 0x1c000})
	/go/src/measure/event.go:1068 +0x174
backend/libs/bus.(*pubSubConsumer).Listen.func1({0x5043420?, 0x3b9d2fad0820?}, 0x3b9d322bd110)
	/go/libs/bus/pubsub_consumer.go:60 +0x36
cloud.google.com/go/pubsub/v2.(*Subscriber).Receive.func2.2.2()
	/go/pkg/mod/cloud.google.com/go/pubsub/v2@v2.5.0/subscriber.go:475 +0x5a
created by cloud.google.com/go/pubsub/v2.(*Subscriber).Receive.func2.2 in goroutine 134
	/go/pkg/mod/cloud.google.com/go/pubsub/v2@v2.5.0/subscriber.go:473 +0x377
```

--- a/backend/api/event/event.go
+++ b/backend/api/event/event.go
@@ -1280,7 +1280,9 @@ func (e EventField) NeedsSymbolication() (result bool) {
 			}
 			result = false
 		case FrameworkDart:
-			if e.Exception.Exceptions[0].Frames[0].InstructionAddr != "" {
+			// Only consider when at least 1 exception &
+			// 1 frame is present.
+			if e.Exception.HasExceptions() && len(e.Exception.Exceptions[0].Frames) > 0 && e.Exception.Exceptions[0].Frames[0].InstructionAddr != "" {
 				result = true
 			}
 		}
@@ -1387,6 +1389,10 @@ func (e Exception) HasNoFrames() bool {
 		}
 		return len(e.Exceptions[0].Frames) == 0
 	case FrameworkDart:
+		// For some cases, exceptions may not be present
+		if !e.HasExceptions() {
+			return true
+		}
 		return len(e.Exceptions[0].Frames) == 0
 	}
 

